context/app/static/js/pages/Workspaces/Workspaces.jsx

```diff
@@ -1,25 +1,46 @@
 import React from 'react';
 
+import EmailIconLink from 'js/shared-styles/Links/iconLinks/EmailIconLink';
 import { useAppContext } from 'js/components/Contexts';
 import WorkspacesTitle from 'js/components/workspaces/WorkspacesTitle';
 import { LightBlueLink } from 'js/shared-styles/Links';
 import WorkspacesAuthenticated from 'js/components/workspaces/WorkspacesAuthenticated';
-
 import { StyledDescription } from './style';
 
 function Workspaces() {
   const { isAuthenticated, isWorkspacesUser } = useAppContext();
+
+  if (!isAuthenticated) {
+    return (
+      <>
+        <WorkspacesTitle />
+        <StyledDescription>
+          The workspaces feature is only available if logged in and is part of the allowed Globus group.{' '}
+          <LightBlueLink href="/login">Log in</LightBlueLink> to view saved workspaces or to begin a new workspace.
+        </StyledDescription>
+      </>
+    );
+  }
+
+  if (!isWorkspacesUser) {
+    return (
+      <>
+        <WorkspacesTitle />
```

Comment:
I don't love that this leads to `<WorkspacesTitle />` on three different lines (since it's always present in all three returns)

My suggestion to avoid this repetition is to split the conditional rendering logic from the non-conditional rendering logic - e.g. if we pull the conditional description logic into a separate (`WorkspacesPage`?) component and let `Workspaces` return a fragment containing the `WorkspacesTitle` and `WorkspacesPage`, we can avoid that bit of repetition. I don't think there'd be any harm in defining the second component in the same file for clarity (since this is in the Pages directory already)

Reply:
I'm going to need more explanation on this one. Let's talk more at standup tomorrow!

Reply:
i.e., hoisting the `WorkspaceTitle` up into a parent component would help prevent needing to wrap every return in a fragment and would reduce repetition of `<WorkspaceTitle />` on the page:

```
function WorkspacesContent() {
  const { isAuthenticated, isWorkspacesUser } = useAppContext();

  if (!isAuthenticated) {
    return (
      <StyledDescription>
        The workspaces feature is only available if you are logged in and part
        of the allowed Globus group.{" "}
        <LightBlueLink href="/login">Log in</LightBlueLink> to view saved
        workspaces or to begin a new workspace.
      </StyledDescription>
    );
  }

  if (!isWorkspacesUser) {
    return (
      <StyledDescription>
        You must be a member of the allowed Globus group to access this feature.
        Email{" "}
        <EmailIconLink variant="body2" email="help@hubmapconsortium.org">
          help@hubmapconsortium.org
        </EmailIconLink>{" "}
        to gain access.
      </StyledDescription>
    );
  }

  return <WorkspacesAuthenticated />;
}

function Workspaces() {
  return (
    <>
      <WorkspacesTitle />
      <WorkspacesContent />
    </>
  );
}
```